src/overlays/actors/ovl_Bg_Astr_Bombwall/z_bg_astr_bombwall.c

```diff
@@ -28,59 +36,194 @@ const ActorInit Bg_Astr_Bombwall_InitVars = {
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+    DynaPolyActor_Init((DynaPolyActor*)&this->dyna.actor, 1);
+    DynaPolyActor_LoadMesh(globalCtx, (DynaPolyActor*)&this->dyna.actor, &D_06002498);
+    temp_a1 = &this->unk160;
+    Collider_InitTris(globalCtx, temp_a1);
+    if (Flags_GetSwitch(globalCtx, this->dyna.actor.params & 0x7F) != 0) {
+        Actor_MarkForDeath(&this->dyna.actor);
+        return;
+    }
+    this->dyna.actor.flags |= 0x10000000;
+    if (Collider_SetTris(globalCtx, temp_a1, &this->dyna.actor, &sTrisInit, &this->unk180) == 0) {
+        Actor_MarkForDeath(&this->dyna.actor);
+        return;
+    }
+    func_80C09ED0(&sTrisInit, &this->dyna.actor.world.pos, &this->dyna.actor.shape.rot, temp_a1);
+    SubS_FillCutscenesList(&this->dyna.actor, &this->unk238, 1);
+    func_80C0A378(this);
+}
+
+void BgAstrBombwall_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BgAstrBombwall* this = THIS;
+    DynaPoly_DeleteBgActor(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+}
+
+void func_80C0A120(BgAstrBombwall* thisx, GlobalContext* globalCtx) {
+    s32 var_s1;
+    Vec3f spC8;
+    Vec3f spBC;
+    s16 var_v0;
+    Vec3f spAC;
+    f32 temp_fv0;
+    Matrix_RotateYS(thisx->dyna.actor.shape.rot.y, MTXMODE_NEW);
+    for (var_s1 = 0; var_s1 < 0x1E; var_s1++) {
+        spC8.x = Rand_Centered() * 140.0f;
+        spC8.y = Rand_ZeroOne() * 200.0f;
+        spC8.z = 0.0f;
+        Matrix_MultVec3f(&spC8, &spBC);
+        Math_Vec3f_Sum(&thisx->dyna.actor.world.pos, &spBC, &spBC);
+        func_800BBFB0(globalCtx, &spBC, 50.0f, 2, Rand_ZeroOne() * 120.0f + 20.0f, Rand_ZeroOne() * 240.0f + 20.0f, 0);
+        spAC.x = Rand_ZeroOne() * 2.5f;
+        spAC.y = (Rand_ZeroOne() * 2.5f) + 1.0f;
+        spAC.z = Rand_ZeroOne() * 2.5f;
+        temp_fv0 = Rand_ZeroOne();
+
+        if (temp_fv0 < 0.2f) {
+            var_v0 = 0x60;
+        } else if (temp_fv0 < 0.6f) {
+            var_v0 = 0x40;
+        } else {
+            var_v0 = 0x20;
+        }
+        EffectSsKakera_Spawn(globalCtx, &spBC, &spAC, &spBC, -0x104, var_v0, 0x14, 0, 0, 0xA, 0, 0, 0x32, -1, 0x267,
+                             &D_06002178);
+    }
+}
+
+void func_80C0A378(BgAstrBombwall* thisx) {
+    thisx->actionFunc = func_80C0A38C;
+}
+
+void func_80C0A38C(BgAstrBombwall* thisx, GlobalContext* globalCtx) {
+    u8 temp_v0;
+
+    temp_v0 = thisx->unk160.base.acFlags;
+    if ((temp_v0 & 2) != 0) {
+        thisx->unk160.base.acFlags = temp_v0 & 0xFFFD;
+        Flags_SetSwitch(globalCtx, thisx->dyna.actor.params & 0x7F);
+        func_80C0A400(thisx, globalCtx);
+        return;
+    }
+    CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &thisx->unk160.base);
+}
+
+void func_80C0A400(BgAstrBombwall* thisx, GlobalContext* globalCtx) {
+    thisx->actionFunc = func_80C0A418;
+}
+
+void func_80C0A418(BgAstrBombwall* thisx, GlobalContext* globalCtx) {
+    if (SubS_StartActorCutscene(&thisx->dyna.actor, thisx->unk238, -1, 0) != 0) {
```

Comment:
```suggestion
    if (SubS_StartActorCutscene(&thisx->dyna.actor, thisx->unk238, -1, SUBS_CUTSCENE_SET_UNK_LINK_FIELDS)) {
```
You can remove the equality operator `!=` and add the macro for the final parameter